test/e2e/recovery_and_restart/recovery_and_restart_test.go

```diff
@@ -0,0 +1,153 @@
+	var kClient k8s.KubeCtl
+	var restClient yunikorn.RClient
+	var oldConfigMap *v1.ConfigMap
+	var sleepRespPod *v1.Pod
+	var dev = "dev" + common.RandSeq(5)
+
+	// Define sleepPod
+	sleepPodConfigs := common.SleepPodConfig{Name: "sleepjob", NS: dev}
+
+	ginkgo.BeforeSuite(func() {
+		// Initializing kubectl client
+		kClient = k8s.KubeCtl{}
+		Ω(kClient.SetClient()).To(gomega.BeNil())
+		// Initializing rest client
+		restClient = yunikorn.RClient{}
+
+		ginkgo.By("Enable basic scheduling config over config maps")
+		var c, err = kClient.GetConfigMaps(configmanager.YuniKornTestConfig.YkNamespace,
+			configmanager.DefaultYuniKornConfigMap)
+		Ω(err).NotTo(gomega.HaveOccurred())
+		Ω(c).NotTo(gomega.BeNil())
+
+		oldConfigMap = c.DeepCopy()
+		Ω(c).Should(gomega.BeEquivalentTo(oldConfigMap))
+
+		// Define basic configMap
+		configStr, err2 := common.CreateBasicConfigMap().ToYAML()
+		Ω(err2).NotTo(gomega.HaveOccurred())
+
+		c.Data[configmanager.DefaultPolicyGroup] = configStr
+		var d, err3 = kClient.UpdateConfigMap(c, configmanager.YuniKornTestConfig.YkNamespace)
+		Ω(err3).NotTo(gomega.HaveOccurred())
+		Ω(d).NotTo(gomega.BeNil())
+
+		ginkgo.By("create development namespace")
+		ns1, err := kClient.CreateNamespace(dev, nil)
+		gomega.Ω(err).NotTo(gomega.HaveOccurred())
+		gomega.Ω(ns1.Status.Phase).To(gomega.Equal(v1.NamespaceActive))
+
+		ginkgo.By("Deploy the sleep pod to the development namespace")
+		sleepRespPod, err = kClient.CreatePod(common.InitSleepPod(sleepPodConfigs), dev)
+		gomega.Ω(err).NotTo(gomega.HaveOccurred())
+		// Wait for pod to move to running state
+		err = kClient.WaitForPodBySelectorRunning(dev,
+			fmt.Sprintf("app=%s", sleepRespPod.ObjectMeta.Labels["app"]),
+			10)
+		gomega.Ω(err).NotTo(gomega.HaveOccurred())
+	})
+
+	ginkgo.It("Restart_Scheduler_Pod", func() {
+		schedulerPodName, err := kClient.GetSchedulerPod()
+		Ω(err).NotTo(gomega.HaveOccurred())
+		err2 := kClient.DeletePod(schedulerPodName, configmanager.YuniKornTestConfig.YkNamespace)
+		Ω(err2).NotTo(gomega.HaveOccurred())
+		err3 := kClient.WaitForPodBySelectorRunning(configmanager.YuniKornTestConfig.YkNamespace, fmt.Sprintf("component=%s", configmanager.YKScheduler), 10)
+		Ω(err3).NotTo(gomega.HaveOccurred())
+	})
+
+	ginkgo.It("Port-forward_Scheduler_Pod_After_Restart", func() {
+		// kill running kubectl port-forward process if it exists
+		kClient.KillPortForwardProcess()
+		// port-forward the scheduler pod
+		err := kClient.PortForwardYkSchedulerPod()
+		Ω(err).NotTo(gomega.HaveOccurred())
+	})
+
+	ginkgo.It("Verify_Pod_Alloc_Props", func() {
+		appsInfo, err := restClient.GetAppInfo(sleepRespPod.ObjectMeta.Labels["applicationId"])
+		gomega.Ω(err).NotTo(gomega.HaveOccurred())
+		gomega.Ω(appsInfo).NotTo(gomega.BeNil())
+		ginkgo.By("Verify the pod allocation properties")
+		gomega.Ω(appsInfo["allocations"]).NotTo(gomega.BeNil())
+		for key, val := range appsInfo {
+			fmt.Println("Key:", key, "=>", "Element:", val)
```

Comment:
Remove this, for loop block on 113-115, and line 116 after resolving issue where allocations are dropped upon scheduler restart and before merging.